### DISCUSSION TITLE
Why does it take so long to start playing?

### DISCUSSION THREAD

**Discussion by ghost** [KRTirtho/spotube#495] [Q&A]
### Is there an existing issue for this?

- [ ] I have searched the existing issues

### Current Behavior

Couldn't you make it play immediately?

### Expected Behavior

Play when I tap play button and not remain loading

### Steps to reproduce

1. I tap play button
2. It doesn't play instantly, it remains loading

### Operating System

Android

### Spotube version

2.7.0

### Installation source

F-Droid (Android)

**Comment by KRTirtho** [comment]
This can happen because of slow network connection or if the track is less-popular and unavailable on YouTube

**Reply by tuwonga** [reply]
this happen in the last days but did not before. I don't think depends on track popularity.

**Reply by KRTirtho** [reply]
@tuwonga follow this thread for more updates https://github.com/KRTirtho/spotube/issues/480#issuecomment-1535705532

**Reply by epic-coder-2976** [reply]
> This can happen because of slow network connection or if the track is less-popular and unavailable on YouTube

Hi, ive the same problem as OP.
Both unpopular and popular songs take quite a bit of time to load, making the experience far less enjoyable. 
And could you clarify why the track's popularity matters in the first place?

**Reply by KRTirtho** [reply]
I'm working on that. This is due to a change in the YouTube API. I'm trying to fix this and will release it when it gets good enough
